src/components/_global/BalAsset/BalAsset.vue

```diff
@@ -1,96 +1,111 @@
 <template>
-  <img
-    v-if="iconSRC && !error"
-    :src="iconSRC"
+  <component
+    class="bal-asset rounded-full inline-block leading-none shadow-sm"
     :style="{
       width: `${size}px`,
-      height: `${size}px`,
-      background: 'white'
+      height: `${size}px`
     }"
-    @error="error = true"
-    class="rounded-full inline-block leading-none shadow-sm"
-  />
-  <Avatar v-else-if="!!address" :address="address" :size="size" />
-  <div
-    v-else
-    class="rounded-full inline-block leading-none shadow-sm overflow-visible"
-    :style="{
-      width: `${size}px`,
-      height: `${size}px`,
-      background: `${noIconColor} !important`
-    }"
-  />
+    :is="rootElement"
+    v-bind="rootElementAttrs"
+  >
+    <img
+      v-if="iconSRC && !error"
+      :src="iconSRC"
+      :style="{
+        background: 'white'
+      }"
+      @error="error = true"
+      class="rounded-full"
+    />
+    <Avatar v-else-if="!!address" :address="address" :size="size" />
+    <div
+      v-else
+      class="rounded-full overflow-visible"
+      :style="{
+        width: `${size}px`,
+        height: `${size}px`,
+        background: `${noIconColor} !important`
```

Comment:
A similar question about using styles in places where the standard use of tailwind seems to fit

We can use `bg-gray-300 dark:bg-gray-700` in class

https://github.com/balancer-labs/frontend-v2/pull/1960/files#r896614263